src/tests/mod.rs

```diff
@@ -1,22 +1,91 @@
 use crate::compiler::AmberCompiler;
 use crate::Cli;
+extern crate test_generator;
+use itertools::Itertools;
+use std::fs;
+use std::path::PathBuf;
 
 pub mod cli;
-pub mod formatter;
 pub mod errors;
-pub mod stdlib;
-pub mod validity;
-
-#[macro_export]
-macro_rules! test_amber {
-    ($code:expr, $result:expr) => {{
-        match AmberCompiler::new($code.to_string(), None, Cli::default()).test_eval() {
-            Ok(result) => assert_eq!(result.trim_end_matches('\n'), $result),
-            Err(err) => panic!("ERROR: {}", err.message.unwrap()),
-        }
-    }};
+pub mod extra;
+pub mod formatter;
+mod stdlib;
+mod validity;
+
+/// compare the output of the given code with the expected output
+pub fn test_amber(code: impl Into<String>, result: impl AsRef<str>) {
+    match AmberCompiler::new(code.into(), None, Cli::default()).test_eval() {
+        Ok(eval_result) => assert_eq!(
+            eval_result.trim_end_matches('\n'),
+            result.as_ref().trim_end_matches('\n')
+        ),
+        Err(err) => panic!("ERROR: {}", err.message.unwrap()),
+    }
 }
 
 pub fn compile_code<T: Into<String>>(code: T) -> String {
-    AmberCompiler::new(code.into(), None, Cli::default()).compile().unwrap().1
+    AmberCompiler::new(code.into(), None, Cli::default())
+        .compile()
+        .unwrap()
+        .1
+}
+
+/// Extracts the output from the comment of amber code
+pub fn extract_output(code: impl Into<String>) -> String {
+    code.into()
+        .lines()
+        .skip_while(|line| !line.starts_with("// Output"))
+        .skip(1) // skip "// Output" itself
+        .take_while(|line| !line.is_empty())
```

Comment:
This will keep taking the lines in this code example:

```
// Output
// Result
echo "Result"
```

As this will treat `echo "Result"` as part of the output comment.

Here is my suggestion:
```suggestion
        .take_while(|line| !line.is_empty() && line.starts_with("//"))
```

Reply:
an improvement of my implementation indeed